### PR TITLE
adding test.env.example to build local CI testing

### PR DESCRIPTION
**Pull request by McKnight-42**
### Description

Adding examples for setting up account to test integration tests locally, based on description from contributing.md being added. to each adapter repo.

- Co author @VersusFacit 
### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-snowflake next" section.

### CODE REVIEW

**Review by VersusFacit — CHANGES_REQUESTED** [review]
Hey, could we alphabetize these? To match the style over in: https://github.com/dbt-labs/dbt-release/blob/40a4fb001f2d2711ffbbd6b518df9f0ce3e232d7/.github/workflows/build.yml#L258

Just makes it easier to read. Unless you can think of a reason not to.

Also, since this is an example, could we _maybe_ add example strings? Or would that be confusing? I know as a user, I like to see examples of what should be in the file fields.

**Comment by VersusFacit** [comment]
@McKnight-42 I like the direction you're headed with the examples. I am thinking what might be most instructive is providing some values that are more or less formatted to what a user might employ. Want to pair up to generate those values? Want to take it on yourself? Just me know either way

**Comment by McKnight-42** [comment]
Sure would love to pair up @VersusFacit

File: test.env.example [hunk 1]
```diff
@@ -0,0 +1,11 @@
+SNOWFLAKE_TEST_ACCOUNT=<account>
```
Comment (VersusFacit) [inline test.env.example:1]: <snowflake_testing_account>

**Comment by VersusFacit** [comment]
Hey Matt, I've been reflecting on this. All this lives in Notion. Should we just tell people internally to consult instructions there where we can share secrets? I'd advocate we don't include this in contributing guides since we aren't handing out relevant DDL that defines these parameters for users. That makes these fields and this setup not all that helpful in pointing users in the right direction.

That's my gut reaction? What's yours? 😄  Let's hash this out in our call tomorrow.

cc: @leahwicz

File: test.env.example (hunk 1 above)
Comment (VersusFacit) [inline test.env.example:1]:
```suggestion
SNOWFLAKE_TEST_ACCOUNT=<your Snowflake account's unique identifier>
```

File: test.env.example [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+SNOWFLAKE_TEST_ALT_DATABASE=<database_name_in_alt_warehouse>
```
Comment (VersusFacit) [inline test.env.example:2]:
```suggestion
SNOWFLAKE_TEST_ALT_DATABASE=<name of database you must create; e.g. DBT_SF_TEST_ALT>
```

File: test.env.example [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+SNOWFLAKE_TEST_ALT_WAREHOUSE=<alt_warehouse_name
```
Comment (VersusFacit) [inline test.env.example:3]:
```suggestion
SNOWFLAKE_TEST_ALT_WAREHOUSE=<name of warehouse you must create; e.g. DBT_SF_TESTING_ALT>
```

File: test.env.example [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+SNOWFLAKE_TEST_DATABASE=<database_name>
```
Comment (VersusFacit) [inline test.env.example:4]:
```suggestion
SNOWFLAKE_TEST_DATABASE=<name of database you must create; e.g. DBT_SF_TEST>
```

File: test.env.example [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+SNOWFLAKE_TEST_OAUTH_CLIENT_ID=<client_id_of_integration
```
Comment (VersusFacit) [inline test.env.example:5]:
```suggestion
# Name of environments variable for OAUTH tokens if you employ this authentication method
SNOWFLAKE_TEST_OAUTH_CLIENT_ID=<client_id_of_integration
```

File: test.env.example [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+SNOWFLAKE_TEST_OAUTH_CLIENT_SECRET=<client_secret_for_integration>
+SNOWFLAKE_TEST_OAUTH_REFRESH_TOKEN=TRUE
+SNOWFLAKE_TEST_PASSWORD=<password>
```
Comment (VersusFacit) [inline test.env.example:8]:
```suggestion
SNOWFLAKE_TEST_PASSWORD=<the password for your test database user>
```

File: test.env.example [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+SNOWFLAKE_TEST_QUOTED_DATABASE=
```
Comment (VersusFacit) [inline test.env.example:9]:
```suggestion
SNOWFLAKE_TEST_QUOTED_DATABASE=<name of a database you must create: e.g. 1-test-database>
```

File: test.env.example [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+SNOWFLAKE_TEST_USER=<username>
```
Comment (VersusFacit) [inline test.env.example:10]:
```suggestion
SNOWFLAKE_TEST_USER=<the name for your test database user>
```

File: test.env.example [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+SNOWFLAKE_TEST_WAREHOUSE=<warehouse_name>
```
Comment (VersusFacit) [inline test.env.example:11]:
```suggestion
SNOWFLAKE_TEST_WAREHOUSE=<name of warehouse you must create; e.g. DBT_SF_TESTING>
```

**Review by VersusFacit — CHANGES_REQUESTED** [review]
This is my stab at describing these fields having just learned what they all actually mean by having to fight through local setup. By all means, alter/correct my wording where think this is unclear. Accept what you like. Let's chat about what seems off.

File: test.env.example [hunk 10]
```diff
@@ -0,0 +1,11 @@
+SNOWFLAKE_TEST_ACCOUNT=<account>
+SNOWFLAKE_TEST_ALT_DATABASE=<database_name_in_alt_warehouse>
+SNOWFLAKE_TEST_ALT_WAREHOUSE=<alt_warehouse_name
+SNOWFLAKE_TEST_DATABASE=<database_name>
+SNOWFLAKE_TEST_OAUTH_CLIENT_ID=<client_id_of_integration
+SNOWFLAKE_TEST_OAUTH_CLIENT_SECRET=<client_secret_for_integration>
```
Comment (VersusFacit) [inline test.env.example:6]:
```suggestion
SNOWFLAKE_TEST_OAUTH_CLIENT_SECRET=client secret of your OAuth client id
```

**Review by VersusFacit — CHANGES_REQUESTED** [review]

File: test.env.example (hunk 1 above)
Comment (leahwicz) [reply test.env.example:1]: The suggested note is confusing to me here. I'm not sure the message you're trying to get across

File: test.env.example (hunk 2 above)
Comment (leahwicz) [reply test.env.example:2]: Nit: "name of existing database"

Comment (leahwicz) [reply test.env.example:2]: Am I correct in saying that "you must create" == "already existing"

**Comment by leahwicz** [comment]
I'm not opposed to being more descriptive here so it's easy to tell what you need to provide. Let's do this for all test.env.example files then

File: test.env.example (hunk 1 above)
Comment (McKnight-42) [reply test.env.example]: took a stab at note, please let me know if its any more clear we can keep working on the most straight forward take for it.

File: test.env.example (hunk 2 above)
Comment (VersusFacit) [reply test.env.example]: That's right. I appreciate that's a more succinct way to capture that.

**Review by VersusFacit — APPROVED** [review]

**Comment by McKnight-42** [comment]
@jtcohen6 @leahwicz @runleonarun. @VersusFacit and I worked on getting this latest version pushed up and were wondering what you all think. Is this how we'd like to present the information as much as we can in the other adapters as well?

**Comment by VersusFacit** [comment]
I advocate for embedded instructions and feel really good about the formatting instructions Leona offered.

File: test.env.example [hunk 11]
```diff
@@ -0,0 +1,31 @@
+# Note: Make sure you have a Snowflake account that is set up so these fields are easy to complete. 
+If you don't have an account set up yet, then take note of these required fields that way, as you're getting set up,
```
Comment (runleonarun) [inline test.env.example:2]: @VersusFacit Just a tiny suggestion to break this sentence up:
```suggestion
If you don't have an account set up yet, then take note of these required fields. When you're getting set up, 
```

**Review by runleonarun — APPROVED** [review]
I made a tiny text suggestion. :shipit:

**Pull request merged**